packages/-ember-data/tests/integration/injection-test.js

```diff
@@ -53,12 +53,19 @@ module('integration/injection eager injections', function (hooks) {
   });
 
   test('did inject', async function (assert) {
-    let foo = store.createRecord('foo');
-    let apple = foo.get('apple');
-    let appleService = this.owner.lookup('service:apple');
-
-    assert.ok(apple, `'model:foo' instance should have an 'apple' property`);
-    assert.ok(apple === appleService, `'model:foo.apple' should be the apple service`);
-    assert.ok(apple instanceof Apple, `'model:foo'.apple should be an instance of 'service:apple'`);
+    // TODO likely this test should instead test that we can use service injections
+    // on models (e.g. that owner is properly setup for it).
+    assert.expectDeprecation(
+      () => {
+        let foo = store.createRecord('foo');
+        let apple = foo.get('apple');
+        let appleService = this.owner.lookup('service:apple');
+
+        assert.ok(apple, `'model:foo' instance should have an 'apple' property`);
+        assert.ok(apple === appleService, `'model:foo.apple' should be the apple service`);
+        assert.ok(apple instanceof Apple, `'model:foo'.apple should be an instance of 'service:apple'`);
+      },
+      { id: 'implicit-injections', count: 1 }
```

Comment:
I think one piece of this RFC is to remove `owner.inject` in 5.0.  Is the intention with listing this deprecation to leave it until then?  Do we have a spot we can explicitly inject the service on the foo model?

Reply:
we don't want to remove our injection until defaultStore is killed off from ember. There is no alternative for folks wanting injection other than for them to manually inject.